File: WebAPI/WebApi/Models/Vendor.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace WebApi.Models
{
    public class Vendor
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public string? BusinessName { get; set; }
        public string? ContactName { get; set; }
        public string? BusinessPhone { get; set; }
        public string? BusinessEmail { get; set; }
        public string? BusinessAddress { get; set; }
        public string? ServiceArea { get; set; } // from the City enum
        public string? ServiceType { get; set; }  // from the ServiceType enum: Catering, Rentals, Entertainment, Staffing.
        public string? CuisineStyle { get; set; }  // froom the CuisineStyle enum: BBQ, Buffet, Beverages, FastFood, StreetFood, Healthy, ...
        public string? CuisineRegion { get; set; } // from the CuisineRegion enum: Italian, Chinese, Indian, Mediterranean, Persian, Caribbean, ...
        public string? Dining { get; set; }  // from the Dining enum: LiveCooking, FoodTruck, Delivery, Pickup
        public string? DietaryType { get; set; }  // from the DietaryType enum: Vegan, Vegetarian, GlutenFree, NutFree, Halal, Kosher
        public string? Description { get; set; }
        public string? ImageUrl { get; set; }

        public string? Website { get; set; }
        public string? Instagram { get; set; }
        public string? Facebook { get; set; }

        public string? Offer { get; set; }
        public DateTime? OfferFrom { get; set; }
        public DateTime? OfferTo { get; set; }

        public bool Featured { get; set; }
        public bool InstantQuoteAvailable { get; set; }
        public bool Favorite { get; set; }

        public double RatingAverage { get; set; }
        public int ReviewCount { get; set; }

        public DateTime CreatedOn { get; set; } = DateTime.Now;

    }
}

File: vendor-user-api/app/Models/Vendor.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

/**
 * @property int $id
 * @property int $user_id
 * @property string $businessName
 * @property string|null $contactName
 * @property string|null $businessPhone
 * @property string|null $businessEmail
 * @property string|null $businessAddress
 * @property string|null $serviceArea
 * @property string|null $serviceType
 * @property string|null $cuisineStyle
 * @property string|null $cuisineRegion
 * @property string|null $dining
 * @property string|null $dietaryType
 * @property string|null $description
 * @property string|null $imageUrl
 * @property string|null $website
 * @property string|null $instagram
 * @property string|null $facebook
 * @property string|null $linkedin
 * @property string|null $offer
 * @property \Illuminate\Support\Carbon|null $offerFrom
 * @property \Illuminate\Support\Carbon|null $offerTo
 * @property bool $featured
 * @property bool $instantQuoteAvailable
 * @property float $ratingAverage
 * @property int $reviewCount
 * @property int $listings
 * @property \Illuminate\Support\Carbon|null $created_at
 * @property \Illuminate\Support\Carbon|null $updated_at
 */
class Vendor extends Model
{
    use HasFactory;

    protected $fillable = [
        'user_id',
        'businessName',
        'contactName',
        'businessPhone',
        'businessEmail',
        'businessAddress',
        'serviceArea',
        'serviceType',
        'cuisineStyle',
        'cuisineRegion',
        'dining',
        'dietaryType',
        'description',
        'imageUrl',
        'website',
        'instagram',
        'facebook',
        'linkedin',
        'offer',
        'offerFrom',
        'offerTo',
        'featured',
        'instantQuoteAvailable',
        'ratingAverage',
        'reviewCount',
        'listings',
    ];

    protected $casts = [
        'offerFrom' => 'datetime',
        'offerTo' => 'datetime',
        'featured' => 'boolean',
        'instantQuoteAvailable' => 'boolean',
    ];

    public function user()
    {
        return $this->belongsTo(User::class);
    }

    public function events()
    {
        return $this->hasMany(Event::class);
    }

    public function reviews()
    {
        return $this->hasMany(Review::class);
    }

    public function favoritedBy()
    {
        return $this->hasMany(FavoriteVendor::class);
    }
}
